libsolidity/interface/StandardCompiler.cpp

```diff
@@ -463,9 +463,18 @@ std::optional<Json::Value> checkOptimizerDetailSteps(Json::Value const& _details
 	{
 		if (_details[_name].isString())
 		{
-			std::string const fullSequence = _details[_name].asString();
+			std::string fullSequence = _details[_name].asString();
+			boost::trim(fullSequence);
 			if (!_runYulOptimizer && !OptimiserSuite::isEmptyOptimizerSequence(fullSequence))
-				return formatFatalError(Error::Type::JSONError, "If Yul optimizer is disabled, only an empty optimizerSteps sequence is accepted.");
+			{
+				std::string errorMessage = "If Yul optimizer is disabled, only an empty optimizerSteps sequence is accepted.";
+				if (fullSequence.empty())
+					errorMessage +=
+						" Note that if an empty or blank string is supplied, the default cleanup sequence is used."
+						" The empty sequence is properly denoted by \":\"."
+					;
```

Comment:
Now that I think about it - this is kinda unnecessary; `isEmptyOptimizerSequence` will return `false` even with whitespace only sequences - so to reduce complexity, it might just make more sense to alter the original message to include `The empty sequence is properly denoted by \":\".`. i.e.
```
If Yul optimizer is disabled, only an empty optimizerSteps sequence is accepted. The empty sequence is properly denoted by \":\".
```

The actual bugfix is basically this `return delimiterCount == 1;`, so no need to over complicate it with the message.

Reply:
I don't think this is over-complicated. This is just an additional test to insert a hint in the error message. I usually don't care about these details, but feels like a defeat to just quit this after we spent some time to get it right (mostly the CLI test tbh).
We can rewrite to make more explicit the white spaces part. I thought that "blank string" would convey that meaning.

Reply:
Also, `Note that if an empty or blank string is supplied, the default cleanup sequence is used.` is technically wrong in this context , since this is part of the error that would be triggered if you were to provide an empty/blank string if the optimizer is disabled, so the default sequence would not be used, since the contract(s) would not compile.

So yeah, maybe better to rephrase it into something like
```
Note that an empty optimizer sequence is is properly denoted by \":\".
```

Reply:
But whatever, I think we spent too much time on this already. 
I'll just add the hint by default.

Reply:
>  What happens is that, when the string is empty, it is assumed that the default cleanup sequence is selected for use, regardless of the optimizer being enabled. The problem fixed by this PR was caused exactly by that.

But this hint will only be shown if the optimizer is disabled right? It's in the 
```
if (!_runYulOptimizer && !OptimiserSuite::isEmptyOptimizerSequence(fullSequence))
```
block.

In any case, don't bother reverting the changes to include the note everywhere, I might be wrong. @cameel can you take a look?

Reply:
Yes, the error message (and the hint) only shows when the optimizer is disabled.
I think the hint would be helpful because the empty/blank string can be confused with the empty optimizer sequence.
But I am fine with it just being appended even if the input is not the empty/blank string. The check would just make it be more specific to that case.

Reply:
Well, having the extra hint is nice since I also think most users will be confused by the difference between an empty string and an empty sequence.

But it does add some redundancy - the check yet again has to make some assumptions about the sequence (e.g. which characters count as whitespace). Honestly, this is trivial enough that I don't particularly mind it if you think it makes things better, but consider whether just always including it the default message wouldn't be a good compromise:

```
If Yul optimizer is disabled, only an empty optimizerSteps sequence is accepted.
Note that an empty sequence is properly denoted by ":".
```

In the current message I'm not sure if the difference between "empty" and "blank" is all that clear.

Reply:
Ok. I had already agreed with the hint by default.
After thinking about it also I think it is better that way because it always clarifies what is the empty optimizer sequence.